File: API/Models/Product.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Models
{
    [Table("Products")]
    public class Product
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string ImageUrl { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }

        public Category Category { get; set; }
        public int? CategoryId { get; set; }
    }
}

File: SPA/src/app/interfaces/product.ts
import { Category } from './category';

export class Product {
    id: number;
    name: string;
    imageUrl: string;
    description: string;
    price: number;
    category: Category;
    categoryId: number;

    constructor(init?: Partial<Product>)
    {
        this.id = 0;
        this.name = '';
        this.imageUrl = '';
        this.description = '';
        this.price = 0;
        this.categoryId = 0;
        this.category = new Category();
    }
}
